src/Tribe/Template/Month.php

```diff
@@ -179,6 +192,48 @@ public function __construct( $args = array() ) {
 		}
 
 		/**
+		 * Indicates if month view cache should be enabled or not.
+		 *
+		 * If the month view cache setting itself is not enabled (or not set) then this
+		 * method will always return false.
+		 *
+		 * In other cases, the default rules are to cache everything except for past
+		 * months and months more than 1yr in the future. This policy can be refined
+		 * or replaced via the 'tribe_events_enable_month_view_cache' filter hook.
+		 *
+		 * @return bool
+		 */
+		protected function enable_month_view_cache() {
+			// Respect the month view caching setting
+			if ( ! tribe_get_option( 'enable_month_view_cache', false ) ) {
+				return false;
+			}
+
+			// Default to always caching the current month
+			if ( ! isset( $this->args[ 'eventDate' ] ) ) {
+				return true;
+			}
+
+			// If the eventDate argument is not in the expected format then do not cache
+			if ( ! preg_match( '/^[0-9]{4}-[0-9]{1,2}$/', $this->args[ 'eventDate' ] ) ) {
+				return false;
+			}
+
+			// If the requested month is in the past, do not cache
+			if ( $this->args[ 'eventDate' ] < date_i18n( 'Y-m' ) ) {
+				return false;
+			}
```

Comment:
Do you think we might want to catch a couple months into the past? I think people would want to click back a couple months and check out events that happened and this could help with that.